### PR TITLE
Typing of attributes (#37)

### LINKED ISSUE #37: Add support for typing EAttributes [5]
EAttributes are currently always typed with EString, but we don't have a mechanism to change that. At some point, we may want to use the properties view for that (#26), but on the short term, it could be useful to introduce a Ctrl+Space command and list all possible EAttribute types (Similar to what is done for selecting nodes through Ctrl + Space commands)

### PR DESCRIPTION
**Pull request by leodennis**
Resolves #37

### CODE REVIEW

**Comment by tortmayr** [comment]
FYI: The CI build is failing because of this:
ERROR: /home/travis/build/eclipsesource/ecore-glsp/client/sprotty-ecore/src/features/edit-label-autocomplete.ts:38:13 - Missing semicolon

**Comment by leodennis** [comment]
@tortmayr Yeah I saw that but I don't get it, the line is `@injectable()` and I've never seen a semicolon after that. Also the lines before and after are all correct.
ddd

**Comment by leodennis** [comment]
I just deleted `@injectable()` as I did not seem to need it...

File: client/sprotty-ecore/src/features/edit-label-autocomplete.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    protected listContainer: HTMLElement;
+    protected currentFocus: number;
+    protected types: string[] = [];
+
+
+    constructor(@inject(TYPES.IActionDispatcher) protected actionDispatcher: GLSPActionDispatcher) {
+        super();
+    }
+
+    protected initializeContents(containerElement: HTMLElement) {
```
Comment (CamilleLetavernier) [inline client/sprotty-ecore/src/features/edit-label-autocomplete.ts:51]: This method is invoked only once, so caching the types here is a little bit risky: any change to the model (Adding/Removing EDataTypes, or renaming an existing EDataType) won't be reflected in the list of available types. This method should probably be called every time on createAutocomplete()

File: client/sprotty-ecore/src/features/edit-label-autocomplete.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+export class ReturnAttributeTypesAction implements ResponseAction {
+    static readonly KIND = 'returnAttributeTypes';
+    kind = ReturnAttributeTypesAction.KIND;
+    types: string[];
+    constructor(public readonly actions: string[], public readonly responseId: string = '') {
+        this.types = actions;
+    }
+}
+
+export class EditLabelUIAutocomplete extends EditLabelUI {
```
Comment (CamilleLetavernier) [inline client/sprotty-ecore/src/features/edit-label-autocomplete.ts:38]: injectable is an annotation; so it should be `@injectable()` (And not `injectable()` like you had in your previous commit). That's probably why the compiler complained

**Review by CamilleLetavernier — CHANGES_REQUESTED** [review]
The changes mostly look good to me; there are just a couple of small things to fix. Thanks!

File: client/sprotty-ecore/src/features/edit-label-autocomplete.ts (hunk 1 above)
Comment (leodennis) [reply client/sprotty-ecore/src/features/edit-label-autocomplete.ts:51]: I overwrote the onBeforeShow method and moved the code there so the types will be fetched just before the edit label will be shown. I added an enum and changed the name and the type was available!

**Review by CamilleLetavernier — APPROVED** [review]

**Pull request merged**